### PR TITLE
Don't create the articles exchange in the dashboard

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Currently the `articles` exchange isn't created until a message is sent, so the queue will miss the first message.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
> Currently the articles exchange isn't created until a message is sent

That's a bug in the article store?

File: dashboard/config/packages/old_sound_rabbit_mq.yaml [hunk 1]
```diff
@@ -9,5 +9,6 @@ old_sound_rabbit_mq:
             exchange_options:
                 name: 'articles'
                 type: fanout
+                declare: false
```
Comment (giorgiosironi) [inline dashboard/config/packages/old_sound_rabbit_mq.yaml:12]: correct not to declare it from here

File: docker-compose.yml [hunk 2]
```diff
@@ -145,7 +145,11 @@ services:
             - APP_SECRET=f6a87c2e031ecadc0d7eb442729435c7
             - DATABASE_URL=pgsql://postgres:@dashboard-postgres:5432/postgres
             - RABBITMQ_URL=amqp://guest:guest@event-bus:5672
-        command: bash -c 'wait-for-port event-bus 5672 && bin/console rabbitmq:consumer events'
+        command: >
```
Comment (giorgiosironi) [inline docker-compose.yml:168]: it doesn't scale to deal with startup in docker-compose configuration, but it's still easier to do it synchronously than to run another "migration"/"setup" container

**Comment by seanwiseman** [comment]
> Currently the articles exchange isn't created until a message is sent, so the queue will miss the first message.

'Might' miss the first message, as the exchange is created and then a message is sent. Between the creation and sending a bind could happen, though granted its unlikely.

> That's a bug in the article store?

The creation/check of the exchange before sending is really just a precaution, perhaps we need to formalize the creation of the exchange to make it more predictable? we can't really have messages being missed.

**Comment by giorgiosironi** [comment]
> really just a precaution, perhaps we need to formalize the creation of the exchange to make it more predictable?

Should happen on bootstrap or before the application is started imho, like for the relational database schema.

**Comment by seanwiseman** [comment]
Agreed. I will add that in.

**Review by giorgiosironi — APPROVED** [review]
The description is out of the date, but the intended behavior is correct

**Pull request merged**